pkg/process/manager.go

```diff
@@ -292,23 +375,89 @@ func (m *Manager) CleanUpDaemonResources(d *daemon.Daemon) {
 	}
 }
 
+func terminateDaemon(d *daemon.Daemon) error {
```

Comment:
Maybe `d.Terminate()`.

Reply:
Looks good. In fact, I am planning to move some other logics and functions to daemon package

Reply:
fixed